gnosis_vpn-lib/src/command.rs

```diff
@@ -46,32 +48,47 @@ pub enum RunMode {
     Init,
     /// after creating safe this state will not be reached again
     PreparingSafe {
-        node_address: String,
+        node_address: Address,
         node_xdai: Balance<XDai>,
         node_wxhopr: Balance<WxHOPR>,
         funding_tool: balance::FundingTool,
     },
     /// Before config generation
     ValueingTicket,
     /// Subsequent service start up in this state and after preparing safe
-    Warmup { hopr_state: String },
+    Warmup { hopr_status: HoprStatus },
     /// Normal operation where connections can be made
-    Running { funding: FundingState, hopr_state: String },
+    Running {
+        funding: FundingState,
+        hopr_status: HoprStatus,
+    },
     /// Shutting down service
     Shutdown,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub enum HoprStatus {
+    Running,
+    Syncing,
+    Starting,
+    Indexing,
+    Initializing,
+    Uninitialized,
+}
```

Comment:
Does hopr-lib not expose this?

Reply:
yes, but it is not serializable nor debug / clonable

Reply:
Raise issues in the hoprnet repo pls.